File: wipro/Aug_8_Part_1/July_30/DemoProject/src/com/java/demo/Employ.java
package com.java.demo;

import java.util.Objects;

public class Employ {

	int empno;
	String name;
	double basic;
	
	
	
	@Override
	public int hashCode() {
		return Objects.hash(basic, empno, name);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Employ other = (Employ) obj;
		return Double.doubleToLongBits(basic) == Double.doubleToLongBits(other.basic) && 
				empno == other.empno
				&& Objects.equals(name, other.name);
	}

	public Employ() {

	}

	public Employ(int empno, String name, double basic) {
		this.empno = empno;
		this.name = name;
		this.basic = basic;
	}

	@Override
	public String toString() {
		return "Employ [empno=" + empno + ", name=" + name + ", basic=" + basic + "]";
	}

	
	
}

File: wipro/July_29_Part_1/Day8/NUnitDemos/Employ.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NUnitDemos
{
    public class Employ
    {
        public int Empno { get; set; }
        public string Name { get; set; }
        public double Basic { get; set; }

        public Employ() { }

        public Employ(int empno, string name, double basic)
        {
            Empno = empno;
            Name = name;
            Basic = basic;
        }
    
        public override string ToString()
        {
            return "Empno " + Empno + " Name " + Name + " Basic " + Basic;
        }
    }
}
